### ISSUE TITLE
Unwanted left shift of target tokens in `get_nll`

### ISSUE THREAD

**Issue by JamesDeAntonis** [huggingface/transformers#8863]
## Environment info

     
- `transformers` version: 3.4
- Platform: ubuntu
- Python version: 3.8
- PyTorch version (GPU?): 1.6
- Tensorflow version (GPU?):
- Using GPU in script?: yes
- Using distributed or parallel set-up in script?:no

## Information

Model I am using (Bert, XLNet ...):

The problem arises when using:
* [ ] the official example scripts: (give details below)
* [x ] my own modified scripts: (give details below)
I am training RAG on the FEVER dataset, trying to generate one token from `['[SUPPORTS]', '[REFUTES]', '[INCONCLUSIVE]'.  My loss function is always zero, I think because of the shift left that occurs in `RagTokenForGeneration.get_nll()`, which I think should only happen if special tokens are included.

The tasks I am working on is:
* [ ] an official GLUE/SQUaD task: (give the name)
* [x] my own task or dataset: (give details below)
Training RAG on FEVER dataset

## To reproduce

1. Simply train `RAGTokenForGeneration` on anything, using only one token with no special tokens.  Loss is zero

**Comment by LysandreJik** [comment]
@patrickvonplaten or @lhoestq might have an idea.

**Comment by patrickvonplaten** [comment]
Hey @JamesDeAntonis yes this is expected. The same behavior would occur for GPT2 if only one token is provided as the labels. You should at least add an EOS token at the end to `labels` (so that you have two labels tokens) to make sure the loss is not zero.

If you cannot do this you will have to fork the repo and manually change the `get_nll()` function.
